packages/lit-dev-content/site/docs/ssr/client-api.md

```diff
@@ -0,0 +1,145 @@
+
+How to re-apply that reactivity client-side depends on whether you are rendering standalone Lit templates or utilizing Lit components.
+
+## Standalone Lit Templates
+"Hydration" for Lit templates is the process of having Lit re-associate the expressions of a Lit templates with the nodes they should update in the DOM as well as adding event listeners. In order to hydrate Lit templates, the `hydrate()` method from the `experimental-hydrate` module is provided in the `lit` package. Before you update a server-rendered container using `render()`, you must first call `hydrate()` on that container using the same template and data that was used to render on the server:
+
+```js
+import {render} from 'lit';
+import {hydrate} from 'lit/experimental-hydrate.js';
+import {myTemplate} from './my-template.js';
+// Initial hydration required before render:
+// (must be same data used to render on the server)
+const initialData = getInitialAppData();
+hydrate(myTemplate(initialData), document.body);
+
+// After hydration, render will efficiently update the server-rendered DOM:
+const update = (data) => render(myTemplate(data), document.body);
+```
+
+## Lit components
+To re-apply reactivity to Lit components, the custom element definition needs to be loaded for them to be upgraded, enabling their lifecycle callbacks, and the templates in the components' shadow roots needs to be hydrated.
+
+Upgrading can be achieved simply by loading the component module that registers the custom element. This can be done by loading a bundle of all the component definitions for a page, or may be done based on more sophisticated heuristics where only a subset of definitions are loaded as needed. To ensure templates in `LitElement` shadow roots are hydrated, load the `lit/experimental-hydrate-support.js` module which installs support for `LitElement` to automatically hydrate itself when it detects it was server-rendered with declarative shadow DOM. This module must be loaded before the `lit` module, and therefore any component modules that import `lit`, is loaded to ensure hydration support is properly installed.
+
+When Lit components are server rendered, their shadow root contents are emitted inside a `<template shadowroot>`, also known as a [Declarative Shadow Root](https://web.dev/declarative-shadow-dom/). Declarative shadow roots automatically attach their contents to a shadow root on the template's parent element when parsed. For browsers that do not yet implement declarative shadow roots, there is a [`template-shadowroot` polyfill](https://github.com/webcomponents/template-shadowroot), whose usage is incorporated in the example below.
+
+Put together, an HTML page that was server rendered and containing Lit components in the main document might look like this:
+
+```js
```

Comment:
This is server code. I'd remove this (it's int he Stackblitz demo) and focus on specific client-side considerations:

- Loading lit/experimental-hydrate-support.js (show HTML snippet)
- Loading DSD polyfill (show inlined code?, show dsd-pending pattern in isolation)
- Show an import of the app code?

Reply:
I did clean up the polyfill usage to not include any hydration since that seems like a valid thing to do for server-rendered things that don't require hydration, but I wasn't sure how to isolate the dsd-pending without it being too redundant. I think it's still better to read now, but we can also drop that and just point to the example with everything in it instead.